Exiled.API/Features/Player.cs

```diff
@@ -777,9 +777,28 @@ public string GroupName
         }
 
         /// <summary>
-        /// Gets the current room the player is in.
+        /// Gets or sets the current room the player is in.
         /// </summary>
-        public Room CurrentRoom => Map.FindParentRoom(GameObject);
+        public Room CurrentRoom
+        {
+            get => Map.FindParentRoom(GameObject);
+            set
+            {
+                PlayerMovementSync.FindSafePosition(value.Position, out Vector3 safePos);
+                switch (value.Type)
```

Comment:
There needs to be more problematic rooms added to this check, for example: 173's room, 079's room, 939's room, etc. You also need to take into account that these rooms can be facing 4 different directions along the x axis, Vector3.back may not always be the safest direction for a particular room.